### ISSUE TITLE
Adding additional wavelength outputs

### ISSUE THREAD

**Issue by bsuther** [geoschem/geos-chem#1594]
Hello, 

Could you please clarify for me what the correct syntax would be to request additional wavelengths to be output for AOD/SSA/Asy using the new geoschem_config.yml file in GEOS-Chem version 14.0.1? I tried changing line 91 from 

```yaml
          aod_wavelengths_in_nm:
            - 550
```
to 
```yaml
          aod_wavelengths_in_nm:
            - 550, 200, 1000
```
which I thought worked because my log file now showed all the requested wavelengths
```console
RRTMG SETTINGS
--------------
AOD output wavelength (nm)  :   550.0
AOD output wavelength (nm)  :   200.0
AOD output wavelength (nm)  :  1000.0
```
I also tried changing that section of the geoschem_config.yml to 

```yaml
     aod_wavelengths_in_nm:
            - 550
            - 200
            - 1000
```
which seemed to produce the same results. However, my GEOSChem.RRTMG.yyyymmdd_0000z.nc4 files still include the AOD at just 550 nm and not the additional wavelengths. I thought that I might need to request the additional wavelengths in HISTORY.rc, so I added 

```console
                              'RadAODWL2_PM          ',
```
to the RRTMG collection in HISTORY.rc (similar to what was mentioned in #1340 ). This does not seem to have worked either, and now I am getting this error, 

```console
===============================================================================
GEOS-Chem ERROR: The "'RadAODWL2_PM" collection attribute did not end with a
":" character!  Please check the HISTORY.rc file.
 -> at Init_DiagList (Headers/diaglist_mod.F90)
 -> ERROR occurred at (or near) line    500 of the HISTORY.rc file
==============================================================================
===========================================================================
GEOS-CHEM ERROR: Error encountered in "Init_DiagList"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
=============================================================================
```
I did look through the documentation and did not find anywhere that specifically addressed this, but if it is there and I missed it, my apologies. 
Thanks

**Comment by yantosca** [comment]
Thanks for writing @bsuther.  The correct form is:
```yaml
     aod_wavelengths_in_nm:
            - 550
            - 200
            - 1000
```
I wonder if there is an issue with the code not picking up the proper value of `WL2` from the YAML file. This is done in `Headers/diaglist_mod.F90` before the state values are initialized (so that this value can be used in GCHP HISTORY):

https://github.com/geoschem/geos-chem/blob/ee8d0eb04d7ad095fe07fb930f28036f256b6709/Headers/diaglist_mod.F90#L288-L305

I can doublecheck that and let you know if there's an issue.

**Comment by yantosca** [comment]
@bsuther, I did a quick test.  In `Headers/diaglist_mod.F90`, I added a print statement and a stop where the diagnostics get read in:
```F90
    ! Read the AOD wavelength in nm for diagnostics
    key   = "operations%rrtmg_rad_transfer_model%aod_wavelengths_in_nm"
    a_str = "UNKNOWN"
    CALL QFYAML_Add_Get( Config, key, a_str, "", RC, dynamic_size=.TRUE. )
    IF ( RC /= GC_SUCCESS ) THEN
       errMsg = 'Error parsing ' // TRIM( key ) // '!'
       CALL GC_Error( errMsg, RC, thisLoc )
       CALL QFYAML_CleanUp( Config          )
       CALL QFYAML_CleanUp( ConfigAnchored  )
       RETURN
    ENDIF
    I = 0
    DO N = 1, SIZE( a_str )
       IF ( TRIM( ADJUSTL( a_str(N) ) ) == "UNKNOWN" ) EXIT
       I = I + 1
       WRITE ( RadWL(I), "(a5)" ) a_str(N)
       RadWL(I) = ADJUSTL( RadWL(I) )
       print*, '### i, radwl(i): ', i, radwl(i)
    ENDDO

    ! Clean up YAML config objects
    CALL QFYAML_CleanUp( Config          )
    CALL QFYAML_CleanUp( ConfigAnchored  )
    STOP
```
Which yielded the following output:
```console
 ### i, radwl(i):            1 550  
 ### i, radwl(i):            2 200  
 ### i, radwl(i):            3 1000 
```
so it is picking up all 3 wavelengths in the `diaglist_mod.F90`. 

I also confirmed that these values are getting read into the `Input_Opt` object in `GeosCore/input_mod.F90`:
```F90
    !------------------------------------------------------------------------
    ! AOD wavelength selection? (You can have up to 3)
    !------------------------------------------------------------------------
    key   = "operations%rrtmg_rad_transfer_model%aod_wavelengths_in_nm"
    a_str = MISSING_STR
    CALL QFYAML_Add_Get( Config, TRIM( key ), a_str,                          &
                         "",     RC,          dynamic_size=.TRUE.            )
    IF ( RC /= GC_SUCCESS ) THEN
       errMsg = 'Error parsing ' // TRIM( key ) // '!'
       CALL GC_Error( errMsg, RC, thisLoc )
       RETURN
    ENDIF

    ! Copy values into Input_Opt
    I = 0
    DO N = 1, SIZE( a_str )
       IF ( a_str(N) == MISSING_STR ) EXIT
       I = I + 1
       Input_Opt%nWvSelect      = I
       Input_Opt%StrWvSelect(I) = TRIM( ADJUSTL( a_str(N) ) )
       Input_Opt%WvSelect(I)    = Cast_and_RoundOff( a_str(N), places=2 )
       print*, '### Input_Opt%wvselect: ',  Input_Opt%WvSelect(I)
    ENDDO
    stop
```
which prints out:
```console
 ### Input_Opt%wvselect:    550.00000000000000     
 ### Input_Opt%wvselect:    200.00000000000000     
 ### Input_Opt%wvselect:    1000.0000000000000     
```
Still looking into other potential issues.

**Comment by bsuther** [comment]
Sure. After I do the run, what info/files would you want to see?

**Comment by yantosca** [comment]
@bsuther: Mostly the log file output.  If you can also do a `ncdump -ct of the GEOSChem.RRTMG.*.nc4` file that would be great.

**Comment by bsuther** [comment]
I think I figured it out. I always try to keep track of changes I have made, so I had added a comment at the end of the lines for the extra fields for WL2 and WL3 in HISTORY.rc. When I remove those comments, then it does create all of the requested variables in the GEOSChem.RRTMG.*.nc4 files. 

However, the _PM variables (RadAOD550nm_PM,RadAsym550nm_PM, RadSSA550nm_PM) are all populated with only 0's. This is true for the standard 550 nm wavelength as well as the additionally requested wavelengths. However, if you look at individual species (i.e. RadAOD550nm_OA or RadAsym550nm_BC) the values are non-zero and seem reasonable.  I think this is an unrelated issue. I created a new run directory where I had **not** changed the .yml or HISTORY.rc file to request additional wavelengths, and the output still had only 0's for those variables. Should I start a new thread about this?

Also, I am not sure if you still want to see it (since it seems like the issue was being caused by my comments in the HISTORY.rc files), but for completeness I will document here the answer to your previous request:
I did compile it with the "cmake . -DCMAKE_BUILD_TYPE=Debug". When I do this it seems to get to where it writes the GEOSChem.RRTMG.*.nc4 file and then crashes without an error message. Here is the log file
[GC.log](https://github.com/geoschem/geos-chem/files/10405069/GC.log)

When I try to do ncdump on the file I just get an error

> $ ncdump -ct GEOSChem.RRTMG.20140701_0000z.nc4
> ncdump: GEOSChem.RRTMG.20140701_0000z.nc4: NetCDF: HDF error

Thanks again for all of your help on this.
